docs/content/design/coordinator.md

```diff
@@ -38,6 +38,44 @@ Balancing Segment Load
 
+Compacting Segments
+-------------------
+
+Each run, the Druid coordinator compacts small segments abutting each other. This is useful when you have a lot of small
+segments which may degrade the query performance as well as increasing the disk usage. Note that the data for an interval
+cannot be compacted across the partitions.
+
+The coordinator first finds the segments to compact together based on the [segment search policy](#segment-search-policy).
+Once it finds some segments, it launches a [compact task](../ingestion/tasks.html#compaction-task) to compact those segments.
+The maximum number of running compact tasks is `max(sum of worker capacity * compactionTaskSlotRatio, 1)`. (See [Dynamic Configuration](../configuration/coordinator.html#dynamic-configuration) for `compactionTaskSlotRatio`.)
+ 
+Compact tasks might fail due to some reasons.
+
+- If the input segments of a compact task are removed or overshadowed before it starts, that compact task fails immediately.
+- If a task of a higher priority acquires a lock for an interval overlapping with the interval of a compact task, the compact task fails.
+
+Once a compact task fails, the coordinator simply finds the segments for the interval of the failed task again, and launches a new compact task in the next run.
+
+To use this feature, you need to set some configurations for dataSources you want to compact.
+Please see [Compaction Config](../configuration/coordinator.html#compaction-config) for details.
+
+### Segment Search Policy
+
+#### Newest Segment First Policy
+
+This policy searches the segments of _all dataSources_ in inverse order of their intervals.
+For example, let me assume there are 3 dataSources (`ds1`, `ds2`, `ds3`) and 5 segments (`seg_ds1_2017-10-01_2017-10-02`, `seg_ds1_2017-11-01_2017-11-02`, `seg_ds2_2017-08-01_2017-08-02`, `seg_ds3_2017-07-01_2017-07-02`, `seg_ds3_2017-12-01_2017-12-02`) for those dataSources.
+The segment name indicates its dataSource and interval. The search result of newestSegmentFirstPolicy is [`seg_ds3_2017-12-01_2017-12-02`, `seg_ds1_2017-11-01_2017-11-02`, `seg_ds1_2017-10-01_2017-10-02`, `seg_ds2_2017-08-01_2017-08-02`, `seg_ds3_2017-07-01_2017-07-02`]. 
+
+Every run, this policy starts searching from the (very latest interval - [skipOffsetFromLatest](../configuration/coordinator.html#compaction-config)).
+This is to handle the late segments ingested to realtime dataSources.
+
+<div class="note caution">
+This policy currently cannot handle the situation when a shard consists of a lot of small segments, thereby its total size exceeds the <a href="../configuration/coordinator.html#compaction-config">targetCompactionSizebytes</a>.
```

Comment:
what does "shard" refer to ? ... a partition-set interval ?

Reply:
Yes.. The above sentence looks not clear, so I changed to ```This policy currently cannot handle the situation when there are a lot of small segments which have the same interval,
and their total size exceeds targetCompactionSizebytes.```.